network/src/transport.rs

```diff
@@ -317,21 +323,33 @@ where
         chain_id: ChainId,
         application_protocols: SupportedProtocols,
     ) -> Self {
-        let mut own_handshake = HandshakeMsg::new(chain_id, network_context.network_id().clone());
-        own_handshake.add(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
+        // build supported protocols
```

Comment:
So, about these comments.  I do appreciate helpful comments, but this may be a little too verbose  on about every 2 lines.  I once lost 20 points on a coding assignment in college from these kinds of comments.

If you want to add comments like this, they should be a level higher.  However, since this whole block basically builds an upgrade_context, it doesn't really need any comments on its own.

I'm a fan of named functions that do pieces, like `build_upgrade_context()`, but that's a personal preference that I don't think aligns with the current code base and gets a little flack especially as an anti-pattern.

Reply:
> I once lost 20 points on a coding assignment in college from these kinds of comments

I wouldn't sweat whatever happened with a bad teacher :P

Reply:
> I'm a fan of named functions that do pieces

I argued against this, we should inline code as much as possible and document the behavior of a function, sure if you understand the function it's useless but it's all about readability and making sure that people reviewing the code can quickly understand the different steps we take